### DISCUSSION TITLE
How do i set priorityClass of my controller pod?

### DISCUSSION THREAD

**Discussion by darklight147** [kubernetes-sigs/kubebuilder#3863] [Support]
Need to give my controller a higher priority Class, is there a marker for it?

**Comment by camilamacedo86** [comment]
Hi @darklight147,

It is an interest question. In this case, you will not find a marker but you can create the PriorityClass resource and sign it to the deployment. 

For example, for you define a priority class for for the Deployment created by your [BusyboxReconciler](https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4-with-deploy-image/internal/controller/busybox_controller.go ), you will need to first create a PriorityClass resource in Kubernetes and then reference that in your Deployment specifications under the PodTemplateSpec, such as:


```go
...
  Template: corev1.PodTemplateSpec{
                ObjectMeta: metav1.ObjectMeta{
                    Labels: ls,
                },
                Spec: corev1.PodSpec{
                    PriorityClassName: "high-my-priority-class", <- HERE add the priority
                    Containers: []corev1.Container{{
                        Image:           image,
                        Name:            "busybox",
                        ImagePullPolicy: corev1.PullIfNotPresent,
                        SecurityContext: &corev1.SecurityContext{
                            RunAsNonRoot:             &[]bool{true}[0],
                            AllowPrivilegeEscalation: &[]bool{false}[0],
                            Capabilities: &corev1.Capabilities{
                                Drop: []corev1.Capability{"ALL"},
                            },
                        },
                    }},
                },
            },
```
```

Then, you also will need to generate the resource such as doing something like beforehand:

```go
    pc := &schedulingv1.PriorityClass{
        ObjectMeta: metav1.ObjectMeta{
            Name: priorityClassName,
        },
        Value:         1000000,
        GlobalDefault: false,
        Description:   "My priority class for the deployment.",
    }
```

At the same way that we create the deployment in: https://github.com/kubernetes-sigs/kubebuilder/blob/2b25adec1ffb8e811a85e8d86357adcf7213bf5b/testdata/project-v4-with-deploy-image/internal/controller/busybox_controller.go#L197-L200

You cannot forget to use the markers to grant the RBAC permissions such as we do for the deployment in: `//+kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete`

I hope that it helps.
I am closing this as sorted out  but if you see that has a need feel free to re-open this one,
